sentry-rails/lib/sentry/rails/action_cable.rb

```diff
@@ -2,20 +2,40 @@ module Sentry
   module Rails
     module ActionCableExtensions
       class ErrorHandler
-        def self.capture(env, transaction_name:, extra_context: nil, &block)
-          Sentry.with_scope do |scope|
-            scope.set_rack_env(env)
-            scope.set_extras(action_cable: extra_context) if extra_context
-            scope.set_transaction_name(transaction_name)
+        class << self
+          def capture(env, transaction_name:, extra_context: nil, &block)
+            Sentry.with_scope do |scope|
+              scope.set_rack_env(env)
+              scope.set_context("action_cable", extra_context) if extra_context
+              scope.set_transaction_name(transaction_name)
+              transaction = start_transaction(env, scope.transaction_name)
+              scope.set_span(transaction) if transaction
 
-            begin
-              block.call
-            rescue Exception => e # rubocop:disable Lint/RescueException
-              Sentry::Rails.capture_exception(e)
+              begin
+                block.call
+                finish_transaction(transaction, 200)
+              rescue Exception => e # rubocop:disable Lint/RescueException
+                Sentry::Rails.capture_exception(e)
+                finish_transaction(transaction, 500)
 
-              raise
+                raise
+              end
             end
           end
+
+          def start_transaction(env, transaction_name)
+            sentry_trace = env["HTTP_SENTRY_TRACE"]
```

Comment:
It's not possible to define HTTP headers on WebSocket connections from Javascript to specify the trace ID. And the Sentry JS library only seems to append the trace id to XHR's. So I'm not sure how to actually use this feature?